client/live.js

```diff
@@ -1,8 +1,37 @@
 var $ = require("jquery");
-var io = require("socket.io-client");
+var SockJS = require("sockjs-client");
 var stripAnsi = require('strip-ansi');
 require("./style.css");
 
+var sock = null;
+var hot = false;
+var currentHash = "";
+
+var newConnection = function(handlers) {
+	sock = new SockJS('/sockjs-node');
+
+	sock.onclose = function() {
+		handlers.close();
+
+		// Try to reconnect.
+		sock = null;
+		setTimeout(function () {
+			newConnection(handlers);
+		}, 2000);
+	};
+
+	sock.onopen = function() {
+		// This callback also fires on a reconnect.
+		handlers.ok();
```

Comment:
On a normal open `ok` comes from the server. However, when the socket gets disconnected and reconnected again, it does not fire an `ok`. So the only way I found to work around this is to call the `ok` function when the socket connection is opened.